src/server/api/index.js

```diff
@@ -735,26 +737,148 @@ function createAPI(app, mountPath, middlewareOpts) {
-        var result = webgmeUtils.getComponentNames(middlewareOpts.gmeConfig.plugin.basePaths);
+        var result = webgmeUtils.getComponentNames(gmeConfig.plugin.basePaths);
         logger.debug('/plugins', {metadata: result});
         res.send(result);
     });
 
+    router.get('/plugins/:pluginId/config', ensureAuthenticated, function (req, res) {
+        var plugin = getPlugin(req.params.pluginId);
+
+        if (plugin instanceof Error) {
+            logger.error(plugin);
+            res.sendStatus(404);
+        } else {
+            res.send(plugin.getDefaultConfig());
+        }
+    });
+
+    router.get('/plugins/:pluginId/configStructure', ensureAuthenticated, function (req, res) {
+        var plugin = getPlugin(req.params.pluginId);
+
+        if (plugin instanceof Error) {
+            logger.error(plugin);
+            res.sendStatus(404);
+        } else {
+            res.send(plugin.getConfigStructure());
+        }
+    });
+
+    router.post('/plugins/:pluginId/execute', ensureAuthenticated, function (req, res) {
+        var resultId = GUID(),
+            pluginContext = {
+                managerConfig: {
+                    project: req.body.projectId,
+                    branchName: req.body.branchName,
+                    commit: req.body.commitHash,
+                    activeNode: req.body.activeNode,
+                    activeSelection: req.body.activeSelection,
+                },
+                pluginConfig: req.body.pluginConfig
+            },
+            workerParameters = {
+                command: middlewareOpts.workerManager.CONSTANTS.workerCommands.executePlugin,
+                webGMESessionId: req.session.id,
+                name: req.params.pluginId,
+                context: pluginContext
+            };
+
+        req.session.save();
+
+        middlewareOpts.workerManager.request(workerParameters, function (err, result) {
+            if (err) {
+                runningPlugins[resultId].status = PLUGIN_CONSTANTS.ERROR;
+                runningPlugins[resultId].err = err;
+            } else {
+                runningPlugins[resultId].status = PLUGIN_CONSTANTS.FINISHED;
+            }
+
+            runningPlugins[resultId].result = result;
+            runningPlugins[resultId].timeoutId = setTimeout(function () {
+                logger.warn('Plugin result timed out: ' + gmeConfig.plugin.serverResultTimeout + '[ms]',
+                    resultId);
+                delete runningPlugins[resultId];
+            }, gmeConfig.plugin.serverResultTimeout);
+        });
+
+        runningPlugins[resultId] = {
+            status: PLUGIN_CONSTANTS.RUNNING,
+            //timeoutId: will be added after plugin finished
+            //result: null,
+            //error: null
+        };
+
+        res.send({resultId: resultId});
+    });
+
+    router.get('/plugins/results/:resultId', ensureAuthenticated, function (req, res) {
```

Comment:
This would mean we cannot have a plugin that is called results, I think.

I suggest to put this functionality to `/plugins/:pluginId/results/:resultId`